Add @bigtest/mocha README

wwilsman opened the pull request:
[rendered](https://github.com/thefrontside/bigtest/blob/ww/mocha-readme/packages/mocha/README.md)

wwilsman commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+describe('clicking my button', () => {
+  // .run() returns a promise that will cause Mocha to wait for the
+  // convergence to resolve before continuing with the assertions
+  beforeEach(() => new Convergence()
```
This has me thinking, should we also wrap `beforeEach` to provide `this.convergence` which could be created with the current mocha timeout?

We could also make it so if you return a `Convergence` it will automatically call `.run()`.

cowboyd replied:
I like the idea of getting your `run()` for free. Is there a way to detect the timeout of the current running test case globally so that you could just:

```js
import { Convergence } from '@bigtest/mocha'
```

And that constructor would detect the currently running testcase and use its timeout to delegate to the constructor of `@bigtest/convergence`

wwilsman replied:
I don't think there is a way to get the current running test's timeout globally. Mocha has `mocha.timeout()` but it's use is for setting the _suite's_ timeout.

The only way I could think of would to update a global `Convergence.timeout` (or some other global) from `it`, which would then be delegated to the constructor of `@bigtest/convergence`.

cowboyd replied:
It just occured to me that we could also just create a new convergence off of the old one, that uses the current timeout.

```js
function beforeEach(setup) {
  return mocha.beforeEach(() => {
    let result = setup.apply(this, arguments);
    if (result instanceof Convergence) {
      return result.timeout(this.timeout).run();
    } else {
      return result;
    }
  });  
}

cowboyd commented on packages/mocha/README.md:
````diff
@@ -1 +1,102 @@
+});
+```
+
+Check out the `@bigtest/convegence` [API
````
`convergence` is mispelled.

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+that any assertions you write using `it` become convergent assertions
+that allow you to easily test asyncronous states.
+
+### Convergent Assertions?
```
I would lead with this. It explains the problem handily, and then we can talk about how it solves the problem. Also, remove the question mark.

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+
+## What is this package?
+
+This package simply wraps Mocha's `it` in a convergence helper so
```
~simply~

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+soon, they will fail; if they run too slow, well then you just have
+slow tests.
+
+Converging on a state simple means asserting against said state until
```
~simple~ s/asserting against said/asserting a state/

wwilsman commented:
Updated based on feedback. Also ran spellcheck and fixed a few mistakes.

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+for more info as to why converging on a desired state is better than
+trying to time it properly._
+
+## What is this package?
```
/What is this package?/How does it work?/

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+
+## Writing Tests
+
+Because convergent assertions are ran repeatedly until they pass, it
```
s/ran/run/

cowboyd commented on packages/mocha/README.md:
````diff
@@ -1 +1,102 @@
+});
+```
+
+### Convergent Hooks?
````
If we want to phrase this as a question, then it has to connect directly to the reader's experience. E.g.

"What if I want to control the timing of my setup code?"

Otherwise, I'd just remove the question mark.

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+
+Because convergent assertions are ran repeatedly until they pass, it
+is _highly recommended_ that you **do not** perform any side-effects
+in your assertions. This will result in the side-effect being ran
```
s/ran/run/

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+Because convergent assertions are ran repeatedly until they pass, it
+is _highly recommended_ that you **do not** perform any side-effects
+in your assertions. This will result in the side-effect being ran
+multiple times in the case when the assertion fails.
```
It isn't just when the assertion fails, if a side-effect takes 500ms to resolve, then we're still running the assertion 50 times. I'd say something like:

> This will result in your side-effect being run multiple, perhaps even hundreds of times.

cowboyd commented on packages/mocha/README.md:
```diff
@@ -1 +1,102 @@
+
+## What is this package?
+
+This package wraps Mocha's `it` in a convergence helper so that any
```
Let's reiterate that this is very low touch difference from the way you write tests. Maybe start with

> For the most part, you write tests in the exact same way that you're used to writing tests with Mocha. The only difference is that this package wraps.....

wwilsman commented:
Second round of feedback integrated. I also realized I completely didn't include `it.always`. It's added now, so maybe one more round of feedback on that section?

cowboyd commented:
Looks great! The only remaining things that I can see to do are:

- [ ] documentation for `it.pause` (or is this too experimental at the moment)
- [ ] linking to `@bigtest/mocha` in the main README.

wwilsman commented:
I don't think `it.pause` is experimental, but I can't figure out a way to test it without making the suite hang. I added it to this README and added `@bigtest/mocha` to the repo README.
